### PR TITLE
[MER-874] Auto Open PR In Browser

### PR DESCRIPTION
**Pull request by alec-stew-art**
added edited code from Go standard library that allows us to open browser on PR creation.

# Testing

- ran `av init` locally on this repository
- generated a test PR on a new branch
- confirmed PR opens up default browser on PR creation for mac running apple silicon
**TODO**: confirm functionality on other operating systems

# Results
Screen Shot 2022-06-17 at 1 48 26 PM

On run of the above command the following window was opened directly in browser
Screen Shot 2022-06-17 at 1 48 13 PM

# Other
Might want to enhance this before merging to allow for the auto-open to not happen if someone doesn't want it to happen.
Maybe a config that is on by default but you can turn off in the `config.yaml`?

### CODE REVIEW

**Comment by alec-stew-art** [comment]
I am not really sure if I have imported anything correctly, and also I still need to do some reading on my end to understand how to locally test the application, but I think something like this general approach works?

File: internal/utils/browser/browser.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+		[]string{"firefox"},
+	)
+	return cmds
+}
+
+// Open tries to open url in a browser and reports whether it succeeded.
+func Open(url string) bool {
+	for _, args := range Commands() {
+		cmd := exec.Command(args[0], append(args[1:], url)...)
+		if cmd.Start() == nil && appearsSuccessful(cmd, 3*time.Second) {
```
Comment (travigd) [inline internal/utils/browser/browser.go:40]: We don't need `appearsSuccessful` below -- that'll just introduce extra latency in a way that I don't think matters for us. Just change to `if err := cmd.Start(); err != nil { ... }`.

Comment (travigd) [reply internal/utils/browser/browser.go:40]: Actually, this function should probably have signature `func Open(url string) error` where we just return the (wrapped) error from `cmd.Start`.

**Comment by alec-stew-art** [comment]
Retested with code changes and we still open a browser window successfully - haven't tested failure message yet but should be fine.

Screen Shot 2022-06-23 at 12 06 31 PM

File: internal/utils/browser/browser.go [hunk 2]
```diff
@@ -1,62 +1,64 @@
...
 	"runtime"
-	"time"
 )
 
-// Commands returns a list of possible commands to use to open a url.
-func Commands() [][]string {
-	var cmds [][]string
+// Open tries to open url in a browser and reports whether it succeeded.
+func Open(url string) error {
+	var args []string = nil
```
Comment (travigd) [inline internal/utils/browser/browser.go:40]: nit: don't need `= nil` here

**Review by travigd — APPROVED** [review]

**Pull request closed without merging**